### DISCUSSION TITLE
ipsec.conf settings from Android .sswan file

### DISCUSSION THREAD

**Discussion by tet-web-dev** [strongswan/strongswan#1887] [Q&A]
I have a .sswan profile created by watchguard vpn. It works perfectly on my Android phone but the watchguard server admin said I can connect with my Ubuntu system that has strongswan installed. I'm getting errors after I kick off the strongswan-starter service and the connection is failing.

**Errors:**
`'username_from_admin' not confirmed by certificate` and `no private key found for 'O=WatchGuard_Technologies, OU=Fireware, CN=Fireware IKE (SN xxx 2023-03-14 13:57:23 UTC) CA'`

**version:** strongSwan U5.9.5/K5.19.0-1025-aws

**.sswan file:**
```
{
    "uuid": "uuid_redacted",
    "name": "XYZ-IKEv2-VPN",
    "type": "ikev2-eap",
    "remote": {
        "addr": "8.x.x.x",
        "cert": "
            <redacted>
            "
    },
    "split-tunneling": {
        "subnets": ["172.x.x.x"]
    },
    "excluded-apps": ["com.google.android.gsf", "com.watchguard.authpoint"]
}
```

**syslog:**
```
ipsec[421754]: 00[NET] using forecast interface ens5
ipsec[421754]: 00[CFG] joining forecast multicast groups: 224.0.0.1,224.0.0.22,224.0.0.251,224.0.0.252,239.255.255.250
ipsec[421754]: 00[CFG] loading ca certificates from '/etc/ipsec.d/cacerts'
ipsec[421754]: 00[CFG] loading aa certificates from '/etc/ipsec.d/aacerts'
ipsec[421754]: 00[CFG] loading ocsp signer certificates from '/etc/ipsec.d/ocspcerts'
ipsec[421754]: 00[CFG] loading attribute certificates from '/etc/ipsec.d/acerts'
ipsec[421754]: 00[CFG] loading crls from '/etc/ipsec.d/crls'
ipsec[421754]: 00[CFG] loading secrets from '/etc/ipsec.secrets'
ipsec[421754]: 00[CFG]   loaded EAP secret for username_from_admin
ipsec[421754]: 00[CFG] loaded 0 RADIUS server configurations
ipsec[421754]: 00[CFG] HA config misses local/remote address
ipsec[421754]: 00[LIB] loaded plugins: charon aesni aes rc2 sha2 sha1 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl fips-prf gmp agent xcbc hmac gcm drbg attr kernel-netlink resolve socket-default connmark forecast farp stroke updown eap-identity eap-aka eap-md5 eap-gtc eap-mschapv2 eap-dynamic eap-radius eap-tls eap-ttls eap-peap eap-tnc xauth-generic xauth-eap xauth-pam tnc-tnccs dhcp lookip error-notify certexpire led addrblock unity counters
ipsec[421754]: 00[LIB] dropped capabilities, running as uid 0, gid 0
ipsec[421754]: 00[JOB] spawning 16 worker threads
ipsec[421750]: charon (421754) started after 60 ms
ipsec[421754]: 05[CFG] received stroke: add connection 'XYZ-IKEv2-VPN'
ipsec[421754]: 05[KNL] 8.x.x.x is not a local address or the interface is down
ipsec[421754]: 05[CFG]   loaded certificate "O=WatchGuard_Technologies, OU=Fireware, CN=Fireware IKE (SN XYZ 2023-03-14 13:57:23 UTC) CA" from '/etc/ipsec.d/certs/rootca.pem'
ipsec[421754]: 05[CFG]   id 'username_from_admin' not confirmed by certificate, defaulting to 'O=WatchGuard_Technologies, OU=Fireware, CN=Fireware IKE (SN XYZ 2023-03-14 13:57:23 UTC) CA'
ipsec[421754]: 05[CFG] added configuration 'XYZ-IKEv2-VPN'
```
**Result from running:** `sudo ipsec up XYZ-IKEv2-VPN COMMAND`
```
ubuntu@ip-xxxx:/$ sudo systemctl restart strongswan-starter
ubuntu@ip-xxxx:/$ sudo ipsec up XYZ-IKEv2-VPN
initiating IKE_SA XYZ-IKEv2-VPN[1] to 8.x.x.x
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
sending packet: from 172.31.x.x[500] to 8.x.x.x[500] (1164 bytes)
received packet: from 8.x.x.x[500] to 172.31.x.x[500] (496 bytes)
parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) V ]
received unknown vendor ID: bf:c2:2e:98:56:ba:99:36:11:c1:1e:48:a6:d2:08:07:a9:5b:ed:b3:93:02:6a:49:e6:0f:ac:32:7b:b9:60:1b:56:6b:34:39:4d:54:49:75:4f:53:34:79:49:45:4a:4f:50:54:59:33:4e:54:67:78:4e:77:3d:3d
selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
local host is behind NAT, sending keep alives
sending cert request for "O=WatchGuard_Technologies, OU=Fireware, CN=Fireware IKE (SN xxx 2023-03-14 13:57:23 UTC) CA"
no private key found for 'O=WatchGuard_Technologies, OU=Fireware, CN=Fireware IKE (SN xxx 2023-03-14 13:57:23 UTC) CA'
establishing connection 'XYZ-IKEv2-VPN' failed
```
**Ipsec.conf**
```
conn XYZ-IKEv2-VPN
     leftsourceip=%config
     auto=add
     ike=aes256gmac-prfsha256-modp2048
     esp=aes256gmac-modp2048
     keyexchange=ikev2
     right=8.x.x.x
     rightid=8.x.x.x
     leftid=username_from_admin
     leftcert=/etc/ipsec.d/certs/rootca.pem
```

**ipsec.secrets:**

`username_from_admin : EAP "password_from_admin"`
**cert file location:** ipsec.d/certs/

**ipsec statusall**
Status of IKE charon daemon (strongSwan 5.9.5, Linux 5.19.0-1025-aws, x86_64):
  uptime: 10 hours, since Sep 05 19:54:36 2023
  malloc: sbrk 3244032, mmap 0, used 1416832, free 1827200
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 0
  loaded plugins: charon aesni aes rc2 sha2 sha1 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl fips-prf gmp agent xcbc hmac gcm drbg attr kernel-netlink resolve socket-default connmark forecast farp stroke updown eap-identity eap-aka eap-md5 eap-gtc eap-mschapv2 eap-dynamic eap-radius eap-tls eap-ttls eap-peap eap-tnc xauth-generic xauth-eap xauth-pam tnc-tnccs dhcp lookip error-notify certexpire led addrblock unity counters
Listening IP addresses:
  172.31.x.x
  20.0.0.1
Connections:
XYZ-IKEv2-VPN:  %any...8.x.x.x  IKEv2
XYZ-IKEv2-VPN:   local:  [O=WatchGuard_Technologies, OU=Fireware, CN=Fireware IKE (SN XXX 2023-03-14 13:57:23 UTC) CA] uses public key authentication
XYZ-IKEv2-VPN:    cert:  "O=WatchGuard_Technologies, OU=Fireware, CN=Fireware IKE (SN XXX 2023-03-14 13:57:23 UTC) CA"
XYZ-IKEv2-VPN:   remote: [8x.x.x] uses public key authentication
XYZ-IKEv2-VPN:   child:  dynamic === dynamic TUNNEL
Security Associations (0 up, 0 connecting):
  none

**Comment by tobiasbrunner** [comment]
Please don't cross-post, see my answer [here](https://serverfault.com/a/1143156/95913).

**Reply by tet-web-dev** [reply]
Thanks, didn't realize the two platforms were so intertwined.

**Reply by tobiasbrunner** [reply]
They are not really, but I get notifications for strongSwan related issues on the Stack Exchange network, so it felt like a cross-posting to me after I answered the question there :)
